python_modules/libraries/dagster-k8s/dagster_k8s/launcher.py

```diff
@@ -212,11 +212,13 @@ def _launch_k8s_job_with_args(
         job_config = container_context.get_k8s_job_config(
             job_image=repository_origin.container_image, run_launcher=self
         )
+        job_image = job_config.job_image
+        if job_image:  # expected to be set
+            self._instance.add_run_tags(
+                run.run_id,
+                {DOCKER_IMAGE_TAG: job_image},
+            )
```

Comment:
If `job_image` is type `Optional[str]` and there no valid conditions in which it should actually be `None` then `assert job_image is not None` is preferable to `if job_image`